io-engine/src/lvs/lvs_lvol.rs

```diff
@@ -865,12 +924,29 @@ impl SnapshotOps for Lvol {
                 })
             }
         }
+
+        let mut attr_descrs: [spdk_xattr_descriptor; SnapshotXattrs::COUNT] =
+            [spdk_xattr_descriptor::default(); SnapshotXattrs::COUNT];
+
+        // Vector to keep allocated CStrings before snapshot  creation
+        // is complete to guarantee validity of attribute buffers
+        // stored inside CStrings.
+        let mut cstrs: Vec<CString> = Vec::new();
```

Comment:
`cstrs` is only for the buffer validation after createsnapshot or can be used as cache in future ? Can it be a hashmap in place of string to fetch data in O(1) ?

Reply:
We don't need to access the content of 'cstrs': we just need to make sure the CStrings in 'cstrs' don't get freed till snapshot is created, because the string buffers inside CStrings are used to store attributes.
So it doesn't matter whether it's a vector or hash - we just need a container. Vector is simpler, IMHO.